cf-agent/files_editline.c

```diff
@@ -208,9 +208,9 @@ Bundle *MakeTemporaryBundleFromTemplate(EvalContext *ctx, Policy *policy, Attrib
             // Get Action operator
             if (strncmp(buffer, "[%CFEngine", strlen("[%CFEngine")) == 0)
             {
-                char op[CF_BUFSIZE], brack[CF_SMALLBUF];
+                char op[CF_BUFSIZE], brack[4];
 
-                sscanf(buffer+strlen("[%CFEngine"), "%1024s %s", op, brack);
+                sscanf(buffer+strlen("[%CFEngine"), "%1024s %3s", op, brack);
 
                 if (strcmp(brack, "%]") != 0)
```

Comment:
What happens here if `brack` is never read from `sscanf()`? Then it contains garbage and this call can overflow.